cow_instrument/ComponentInfo.h

```diff
@@ -0,0 +1,128 @@
+#ifndef COMPONENT_INFO_H
+#define COMPONENT_INFO_H
+
+#include <memory>
+#include <vector>
+#include <stdexcept>
+#include <string>
+#include <sstream>
+#include <cmath>
+
+#include "ComponentProxy.h"
+#include "cow_ptr.h"
+#include "Detector.h"
+#include "IdType.h"
+#include "L1s.h"
+#include "L2s.h"
+#include "MaskFlags.h"
+#include "MonitorFlags.h"
+#include "Path.h"
+#include "PathComponent.h"
+#include "PathFactory.h"
+#include "Spectrum.h"
+
+/**
+ * ComponentInfo type. Provides meta-data an behaviour for working with a
+ * FlatTree
+ * at the component level. Meta-data is provided per component. Add detectors
+ * are components.
+ */
+template <typename InstTree> class ComponentInfo {
+public:
+  template <typename InstSptrType>
+  explicit ComponentInfo(InstSptrType &&instrumentTree);
+
+  Eigen::Vector3d position(size_t componentIndex) const;
```

Comment:
So detectors are components and have a component index? This implies that `DetectorInfo` does not store positions, and in turn `SpectrumInfo` cannot share a position vector with `DetectorInfo`?